compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -618,12 +618,15 @@ public STNode resumeParsing(ParserRuleContext context, Object... args) {
                 return parseArgsBindingPatternEnd();
             case TABLE_ROW_END:
                 return parseTableRowEnd();
+            case LIST_BP_OR_LIST_CONSTRUCTOR_MEMBER:
+                return parseListBindingPatternOrListConstructorMember(peek().kind);
+            case TUPLE_TYPE_DESC_OR_LIST_CONST_MEMBER:
+                return parseTupleTypeDescOrListConstructorMember(peek().kind, (STNode) args[0]);
```

Comment:
we haven't used peek() inside the resumeParsing method. Let's stick to a single pattern.

Reply:
alright